src/Interpreters/TreeOptimizer.cpp

```diff
@@ -740,8 +743,8 @@ void TreeOptimizer::apply(ASTPtr & query, TreeRewriterResult & result,
     if (!select_query)
         throw Exception("Select analyze for not select asts.", ErrorCodes::LOGICAL_ERROR);
 
-    if (settings.optimize_functions_to_subcolumns && result.storage_snapshot && result.storage->supportsSubcolumns())
-        optimizeFunctionsToSubcolumns(query, result.storage_snapshot->metadata);
+    if (settings.optimize_functions_to_subcolumns)
```

Comment:
Why `storage_snapshot` is not null, but `storage` is null ? In other places we assume that they both not null.

Reply:
You are right. I found the root cause.